File: android/app/src/main/java/com/soar/tracker/data/repository/AuthRepository.kt
package com.soar.tracker.data.repository

import com.soar.tracker.data.api.LoginRequest
import com.soar.tracker.data.api.SoarApi
import com.soar.tracker.data.local.TokenManager

class AuthRepository(
    private val tokenManager: TokenManager,
) {
    suspend fun login(api: SoarApi, email: String, password: String): Result<String> {
        return try {
            val response = api.login(LoginRequest(email, password))
            if (response.isSuccessful) {
                val body = response.body() ?: return Result.failure(Exception("Empty response"))
                tokenManager.saveToken(body.token)
                val name = listOfNotNull(body.user.firstName, body.user.lastName)
                    .joinToString(" ")
                    .ifBlank { body.user.email ?: "User" }
                tokenManager.saveUserName(name)
                Result.success(name)
            } else {
                val errorBody = response.errorBody()?.string() ?: "Login failed"
                Result.failure(Exception(errorBody))
            }
        } catch (e: Exception) {
            Result.failure(e)
        }
    }

    fun logout() {
        tokenManager.clear()
    }

    fun isLoggedIn(): Boolean = tokenManager.isLoggedIn()

    fun getUserName(): String? = tokenManager.getUserName()
}

File: ios/SOARTracker/Data/Repository/AuthRepository.swift
import Foundation

class AuthRepository {
    private let tokenManager = TokenManager.shared

    func login(serverURL: String, email: String, password: String) async throws -> String {
        let api = SoarAPI(baseURL: serverURL)
        let response = try await api.login(email: email, password: password)

        // Store credentials
        tokenManager.token = response.token
        tokenManager.serverURL = serverURL

        // Build display name
        let user = response.user
        let name: String
        if let first = user.firstName, let last = user.lastName, !first.isEmpty {
            name = "\(first) \(last)".trimmingCharacters(in: .whitespaces)
        } else {
            name = user.email
        }
        tokenManager.userName = name

        return name
    }

    func requestPasswordReset(serverURL: String, email: String) async throws {
        let api = SoarAPI(baseURL: serverURL)
        try await api.requestPasswordReset(email: email)
    }

    func logout() {
        tokenManager.clear()
    }

    func createAuthenticatedAPI() -> SoarAPI? {
        guard let token = tokenManager.token else { return nil }
        return SoarAPI(baseURL: tokenManager.serverURL, token: token)
    }
}
